tests/test_views.py

```diff
@@ -3605,10 +3614,193 @@ def test_confirmation_link_registers_user(self):
         assert_true(user.is_registered)
 
 
-class TextExternalAuthViews(OsfTestCase):
+class TestAuthLoginAndRegisterLogic(OsfTestCase):
+
+    def setUp(self):
+        super(TestAuthLoginAndRegisterLogic, self).setUp()
+        self.no_auth = Auth()
+        self.user_auth = AuthUserFactory()
+        self.auth = Auth(user=self.user_auth)
+
+    def test_osf_login_and_register(self):
+
+        # login: user with auth
```

Comment:
Ideally each of these should be individual tests. That way, if the tests fail, it will be very specific what the problem is. It also will give all of the errors from a failure rather than stopping when it hits the first.